packages/odyssey-react-mui/src/MenuButton.tsx

```diff
@@ -93,23 +82,27 @@ export type MenuButtonProps = {
    * The tooltip text for the Button if it's icon-only
    */
   tooltipText?: string;
-} & (
-  | {
-      ariaLabel?: string;
-      ariaLabelledBy?: string;
-      buttonLabel: string;
-    }
-  | {
-      ariaLabel: string;
-      ariaLabelledBy?: string;
-      buttonLabel?: undefined | "";
-    }
-  | {
-      ariaLabel?: string;
-      ariaLabelledBy: string;
-      buttonLabel?: undefined | "";
-    }
-) &
+} & Pick<
+  AriaAttributeProps,
+  "ariaDescribedBy" | "ariaLabel" | "ariaLabelledBy"
+> &
+  (
+    | {
+        ariaLabel?: string;
+        ariaLabelledBy?: string;
+        buttonLabel: string;
+      }
+    | {
+        ariaLabel: string;
+        ariaLabelledBy?: string;
+        buttonLabel?: undefined | "";
+      }
+    | {
+        ariaLabel?: string;
+        ariaLabelledBy: string;
+        buttonLabel?: undefined | "";
+      }
+  ) &
```

Comment:
Is this what we want here? I'd prefer to always reference the `Pick` if possible when pulling these props.

```suggestion
} & Pick<
  AriaAttributeProps,
  "ariaDescribedBy"
> &
  (
    | (
      Partial<
        Pick<
          AriaAttributeProps,
          "ariaLabel" | "ariaLabelledBy"
        >
      >
      & Pick<
        AriaAttributeProps,
        "buttonLabel"
      >
    )
    | (
      Partial<
        Pick<
          AriaAttributeProps,
          "ariaLabelledBy" | "buttonLabel"
        >
      >
      & Pick<
        AriaAttributeProps,
        "ariaLabel"
      >
    )
    | (
      Partial<
        Pick<
          AriaAttributeProps,
          "ariaLabel" | "buttonLabel"
        >
      >
      & Pick<
        AriaAttributeProps,
        "ariaLabelledBy"
      >
    )
  ) &
```

Reply:
This feels far more confusing to me. I don't mind the first implementation personally